apps/emqx/src/emqx_persistent_message_ds_replayer.erl

```diff
@@ -66,10 +66,9 @@
 
 -opaque inflight() :: #inflight{}.
 
--type reply_fun() :: fun(
-    (seqno(), emqx_types:message()) ->
-        emqx_session:replies() | {_AdvanceSeqno :: false, emqx_session:replies()}
-).
+-type replies() :: reply() | [replies()].
```

Comment:
Nit: this seems to imply that `[[[[fun(.....)]]]] :: replies()`.  Is that right?

If so, `publish/3` might eventually return `{[[[emqx_session:replies()]]], _}`, which might not be expected by places that call `publish/3`?

Reply:
Should now be fixed after e02e635.